src/main/java/de/hysky/skyblocker/skyblock/ChestValue.java

```diff
@@ -138,19 +166,42 @@ private static Text getDungeonChestProfit(GenericContainerScreenHandler handler,
 				profit -= kismetPriceData.leftDouble();
 			}
 
-			return Text.literal(titleString).append(getProfitText((long) profit, hasIncompleteData));
+			return getProfitText((long) profit, hasIncompleteData);
 		} catch (Exception e) {
 			LOGGER.error("[Skyblocker Profit Calculator] Failed to calculate dungeon chest profit! ", e);
 		}
 
-		return title;
+		return null;
 	}
 
-	private static Text getChestValue(GenericContainerScreenHandler handler, Text title, String titleString) {
+	private static @Nullable Text getChestValue(GenericContainerScreenHandler handler, boolean minion) {
 		try {
 			double value = 0;
 			boolean hasIncompleteData = false;
-			List<Slot> slots = handler.slots.subList(0, handler.getRows() * 9);
+			List<Slot> slots;
+			if (minion) {
+				slots = handler.slots.subList(0, handler.getRows() * 9).stream().filter(slot -> {
+					int x = slot.id % 9;
+					int y = slot.id / 9;
+					return x > 2 && x < 8 && y > 1 && y < 5;
+				}).toList();
+				ItemStack stack;
+				String coinsLine;
+				if (handler.slots.size() > 28 &&
+						(stack = handler.slots.get(28).getStack()).isOf(Items.HOPPER) &&
+						(coinsLine = ItemUtils.getLoreLineIf(stack, s -> s.contains("Held Coins:"))) != null) {
+
+					String source = coinsLine.split(":")[1];
+					try {
+						value += DecimalFormat.getNumberInstance(java.util.Locale.US).parse(source.trim()).doubleValue();
+					} catch (ParseException e) {
+						LOGGER.warn("[Skyblocker] Failed to parse {}", source);
+					}
+
+				}
```

Comment:
I would separate all of this into a separate method, like `getMinionSlots`.

Reply:
that returns the slots or the coins 🤔

Reply:
The slots seem easier and nicer.